src/primitives/certificate.cpp

```diff
@@ -262,6 +295,31 @@ CAmount CScCertificate::GetValueOfChange() const
     return nValueOut;
 }
 
+libzendoomc::ScFieldElement calculateCertDataHash(const CScCertificate& cert) {
```

Comment:
We can put these 2 methods to the `proofverifier` file or create a new one in `sc` dir.